api/apps/geoprocessing/src/modules/scenario-planning-units-inclusion/scenario-planning-units-inclusion-processor.ts

```diff
@@ -1,12 +1,108 @@
 import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Brackets, In, Repository, WhereExpression } from 'typeorm';
+import { MultiPolygon, Polygon } from 'geojson';
+import { flatten } from 'lodash';
 import { Job } from 'bullmq';
+
 import { WorkerProcessor } from '@marxan-geoprocessing/modules/worker';
 import { JobInput } from '@marxan-jobs/planning-unit-geometry';
+import {
+  LockStatus,
+  ScenariosPlanningUnitGeoEntity,
+} from '@marxan/scenarios-planning-unit';
 
 @Injectable()
 export class ScenarioPlanningUnitsInclusionProcessor
   implements WorkerProcessor<JobInput, true> {
-  async process(_job: Job<JobInput, true>): Promise<true> {
+  constructor(
+    @InjectRepository(ScenariosPlanningUnitGeoEntity)
+    private readonly scenarioPlanningUnitsRepo: Repository<ScenariosPlanningUnitGeoEntity>,
+  ) {}
+
+  async process(job: Job<JobInput, true>): Promise<true> {
+    const scenarioId = job.data.scenarioId;
+    const includeGeo = job.data.include?.geo;
+    const excludeGeo = job.data.exclude?.geo;
+
+    const geometriesIdsToInclude: string[] = [];
+    const geometriesIdsToExclude: string[] = [];
+
+    if (includeGeo) {
+      const targetGeometries = flatten(
+        includeGeo.map((collection) => collection.features),
+      ).map((feature) => feature.geometry);
+      geometriesIdsToInclude.push(
+        ...(
+          await this.getIntersectedGeometriesFor(scenarioId, targetGeometries)
+        ).map(({ pu_geom_id }) => pu_geom_id),
```

Comment:
please remember about the limit of arguments
~100k

not sure about the number of it, but with hundreds of thousands stack size may be exceeded

Reply:
maybe worth setting a sensible hard limit and either just log if we hit it (which may still be followed by an overflow) or outright reject the job, so that if we start seeing some of this failures we can think about a different strategy such as splitting the task into smaller batches?

Reply:
I agree at some point in future it may matter, does not seem like worth digging at this very moment.